test/crud-with-options.test.js

```diff
@@ -536,6 +536,54 @@ describe('crud-with-options', function() {
 
   });
 
+  describe('upsertWithWhere', function() {
+
+    beforeEach(seed);
+
+    it('should allow upsertWithWhere(data, cb)', function(done) {
+      User.upsertWithWhere({ name: 'John Smith' }, function() {
+        User.find({ where: { name: 'John Lennon' }}, function(err, data) {
+          should.not.exist(err);
+          data.length.should.equal(1);
+          User.find({ where: { name: 'John Smith' }}, function(err, data) {
+            should.not.exist(err);
+            data.length.should.equal(0);
+            done();
+          });
+        });
+      });
+    });
+
+    it('should allow upsertWithWhere(where, data, cb)', function(done) {
+      User.upsertWithWhere({ where: { name: 'John Lennon' }}, { name: 'John Smith' }, function(err) {
+        should.not.exist(err);
```

Comment:
As mentioned else where use `if (err) return done(err);` for better error reporting even though you see we are using `should.not.exist(err);` in our previous code; `should.not.exist(err);` is bad we should avoid it.